app/src/main/java/com/example/springprojectstationbuild/MainActivity.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.springprojectstationbuild
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import androidx.compose.runtime.*
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.platform.LocalContext
+
+
+
+class MainActivity : AppCompatActivity() {
+
+    private val viewModel: MainViewModel by viewModel<MainViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            viewModel.fetchComputerComponents()
+            val components by viewModel.components.observeAsState(initial = emptyList())
+            R.layout.settings_activity
+
+        }
+    }
+}
+@Composable
+fun mainButtons() {
```

Comment:
Composable functions diverge from the common practice of lowerCamelCase, and should be nouns, that are PascalCase, believe it or not.